base/deprecated.jl

```diff
@@ -1298,6 +1298,10 @@ end
      end
 end
 
+# #19635
+@deprecate zeros{T <: Number}(::Type{T}, arr::Range)  zeros(T, length(arr))
```

Comment:
I could restore `zeros(Int, [1,2,3])` with a depwarn, but `zeros(Int, [1,2,3])` was buggy, see #19265 and I have a test that this case throws an error. So a depwarn is awkward.